src/Playwright.Tests/TaskUtils.cs

```diff
@@ -58,6 +58,13 @@ public static async Task<(T1, T2, T3)> WhenAll<T1, T2, T3>(Task<T1> task1, Task<
             return (task1.Result, task2.Result, task3.Result);
         }
 
+        public static async Task<(T1, T2, T3, T4)> WhenAll<T1, T2, T3, T4>(Task<T1> task1, Task<T2> task2, Task<T3> task3, Task<T4> task4)
```

Comment:
drive-by: addition: TaskUtil.WhenAll was not working for 4 promises, this makes it now work.